src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -326,6 +328,9 @@ public IEnumerable<ITagSpan<IClassificationTag>> GetTags(NormalizedSnapshotSpanC
                     AddClassifiedSpansForPreviousTree(
                         classificationService, span, lastSnapshot, lastDocument, classifiedSpans);
                 }
+
+                // Ensure the syntax tree stays alive for as long as we're doing the processing.
+                GC.KeepAlive(lastSyntaxTree);
```

Comment:
❗ This use of `GC.KeepAlive` is not necessary. `Document._syntaxTreeResultTask` always holds a strong reference to the `SyntaxTree` for the life of the `Document` instance.

Reply:
That's true. But it depends on an impl detail. If we can doc this though, of feel more comfortable removing this line.

Reply:
Why not just remove the earlier call?

Reply:
> If we can doc this though

Actually feel free to do this. `Document.GetSyntaxTreeAsync` often returns a `Task<T>` synchronously, but it's a public API so we can't change it to `ValueTask<T>`. It would be beneficial to users to document that the operation caches the return value so future calls are not expensive.